pkg/server/plugin/upstreamca/awssecret/awssecret_test.go

```diff
@@ -0,0 +1,182 @@
+package awssecret
+
+import (
+	"context"
+	"encoding/json"
+	"encoding/pem"
+	"io/ioutil"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/secretsmanager"
+	"github.com/aws/aws-sdk-go/service/secretsmanager/secretsmanageriface"
+	spi "github.com/spiffe/spire/proto/common/plugin"
+	"github.com/spiffe/spire/proto/server/upstreamca"
+	"github.com/stretchr/testify/suite"
+)
+
+var (
+	ctx = context.Background()
+)
+
+const (
+	config = `{
+	"ttl":"1h",
+	"key_file_arn":"key",
+	"cert_file_arn":"cert",
+	"access_key_id":"keyid",
+	"region":"us-west-2",
+	"secret_access_key":"accesskey",
+	"use_fake_secretsmanager":"yes"
+}`
+	trustDomain = "example.com"
+)
+
+type AWSSecretSuite struct {
+	suite.Suite
+
+	client        secretsmanageriface.SecretsManagerAPI
+	awsUpstreamCA upstreamca.Plugin
+}
+
+func (s *AWSSecretSuite) SetupTest() {
+	var (
+		err error
+	)
+	s.awsUpstreamCA, err = s.newAWSUpstreamCA()
+	s.Require().Nil(err)
+}
+
+func TestAWSSecret(t *testing.T) {
+	suite.Run(t, new(AWSSecretSuite))
+}
+
+func (as *AWSSecretSuite) TestConfigureNoGlobal() {
+	a := New(newFakeSecretsManagerClient)
+	req := new(spi.ConfigureRequest)
+	resp, err := a.Configure(nil, req)
+	as.Require().NotNil(err)
```

Comment:
`NoError` is the correct method to check for absence of an error (leads to slightly clearer messaging on failure)